pwndbg/gdblib/heap/__init__.py

```diff
@@ -124,3 +124,48 @@ def resolve_heap(is_first_run: bool = False) -> None:
             )
     else:
         current = pwndbg.gdblib.heap.ptmalloc.DebugSymsHeap()
+
+
+def heap_freebin_address_lookup(addr: int) -> str | None:
+    """
+    Return a string representing the free bin that an address belongs to.
+    None if doesn't belong to a bin.
+
+    Only supports GLIBC.
+    """
+    allocator = pwndbg.gdblib.heap.current
+    if not isinstance(allocator, pwndbg.gdblib.heap.ptmalloc.GlibcMemoryAllocator):
+        return None
+
+    try:
+        heap = pwndbg.gdblib.heap.ptmalloc.Heap(addr)
+    except Exception:
+        return None
+
+    if heap.arena is None:
+        return None
+
+    for chunk in heap:
+        if addr in chunk:
+            arena = chunk.arena
+            if arena:
```

Comment:
I don't think `chunk.arena` will ever be different to `heap.arena` under normal circumstances.

In fact I believe `Chunk` does `self.heap.arena` internally.